ArduCopter/Copter.cpp

```diff
@@ -445,6 +445,40 @@ bool Copter::has_ekf_failsafed() const
     return failsafe.ekf;
 }
 
+// get target location (for use by scripting)
+bool Copter::get_target_location(Location& target_loc)
+{
+    switch (flightmode->mode_number()) {
+    case Mode::Number::RTL:
+    case Mode::Number::SMART_RTL:
+    case Mode::Number::AVOID_ADSB:
+    case Mode::Number::GUIDED:
+    case Mode::Number::AUTO:
+    case Mode::Number::FOLLOW:
+        return flightmode->get_wp(target_loc);
+        break;
+    default:
+        break;
+    }
+    return false;
+}
+
+/*
+  update_target_location() acts as a wrapper for set_target_location
+ */
+bool Copter::update_target_location(const Location &old_loc, const Location &new_loc)
```

Comment:
I know this is just a copy from plane, but I'm not really sure what passing the old location in here is really doing for us?

Reply:
@IamPete1 Thank you for review this PR. I believe this feature was added for ship landing. I think this functionality requires continually updating the target position using the offset from the last target. Just a guess..
By the way, I PR this because I want to try ship landing with rover and copter.

Reply:
I agree, we already have get_target_location and set_target_location bindings so why would we need update_target_location()?

Reply:
@rmackay9 Thank you for review! From what I understand, set_target_location can be specified with any location and alt frame, whereas update_target_location seems to be checking if it is continuous with the same alt frame as the previous target. For use cases like ship landings.